### PR TITLE
Add string slice flag

### LINKED ISSUE #30: Add Flag List support
Add support for list flags as in arguments.

### PR DESCRIPTION
**Pull request by chroblert**
**add StringSlice flag**

`f.StringSlice("H","Headers",[]string{},"string slice")`
Enter multiple identical flags in the command line, and use c.Flags.StringSlice() to get a []string.
**eg**: 
```
enter: flags -H var1=val1 -H var2=val2,
get:    [var1=val1 var2=val2]
```
image
This StringSlice flag is suitable for sending http request messages.
**eg**:
`Repeat -u test.com -H "Accept: application/json, text/javascript, */*; q=0.01" -H "Origin: https://test.com"`

closes #30

### CODE REVIEW

**Comment by skaldesh** [comment]
related to #30

**Comment by skaldesh** [comment]
Could you clean up the code? You have a lot of stuff simply commented out

**Comment by chroblert** [comment]
deleted useless comment on flags.go,flagmap.go

File: sample/full/cmd/flags.go [hunk 1]
```diff
@@ -50,6 +51,7 @@ func init() {
 			fmt.Println("uint     ", c.Flags.Uint("uint"))
 			fmt.Println("uint64   ", c.Flags.Uint64("uint64"))
 			fmt.Println("float    ", c.Flags.Float64("float"))
+			fmt.Println("StringSLice",c.Flags.StringSlice("Headers"))
```
Comment (r0l1) [inline sample/full/cmd/flags.go:54]: Please fix typo.

**Review by r0l1 — CHANGES_REQUESTED** [review]

File: sample/full/cmd/flags.go (hunk 1 above)
Comment (skaldesh) [reply sample/full/cmd/flags.go:54]: and make sure to run go fmt, there is a missing whitespace between the arguments

File: flags.go [hunk 2]
```diff
@@ -202,6 +202,57 @@ func (f *Flags) String(short, long, defaultValue, help string) {
...
+			}
+
+		},
+		func(flag, equalVal string, args []string, res FlagMap) ([]string, bool, error) {
+			if !f.match(flag, short, long) {
+				return args, false, nil
+			}
+			if len(equalVal) > 0 {
+				if res[long] != nil {
+					res[long].Value = append(res[long].Value.([]interface{}), trimQuotes(equalVal))
```
Comment (skaldesh) [inline flags.go:224]: this branch is unnecessary. do a `if res[long] == nil` check and initialize the flagmapitem there. then afterwards, add the equalval

File: flags.go [hunk 3]
```diff
@@ -202,6 +202,57 @@ func (f *Flags) String(short, long, defaultValue, help string) {
...
+					}
+					res[long].Value = append(res[long].Value.([]interface{}), trimQuotes(equalVal))
+					res[long].IsDefault = false
+				}
+				return args, true, nil
+			}
+			if len(args) == 0 {
+				return args, false, fmt.Errorf("missing string value for flag: %s", flag)
+			}
+			if res[long] != nil {
```
Comment (skaldesh) [inline flags.go:239]: same thing here. remove the code duplication

File: flagmap.go [hunk 4]
```diff
@@ -64,6 +64,23 @@ func (f FlagMap) String(long string) string {
...
 }
 
+// StringSlice returns the given flag value as string slice.
+// Panics if not present. Flags must be registered.
+func (f FlagMap) StringSlice(long string) []string {
+	i := f[long]
+	if i == nil {
+		panic(fmt.Errorf("missing flag value: flag '%s' not registered", long))
+	}
+	s := make([]string, 0)
```
Comment (skaldesh) [inline flagmap.go:74]: cache the length of `i.Value.([]interface{})` and initialize the slice with the length.

File: flagmap.go [hunk 5]
```diff
@@ -64,6 +64,23 @@ func (f FlagMap) String(long string) string {
...
+	if len(i.Value.([]interface{})) == 0 {
+		return s
+	}
+	for _, v := range i.Value.([]interface{}) {
```
Comment (skaldesh) [inline flagmap.go:78]: when the slice above is initialized to its proper length already, use the loop index here to directly assign the value, avoiding `append` completely

**Review by skaldesh — CHANGES_REQUESTED** [review]

File: sample/full/cmd/flags.go (hunk 1 above)
Comment (chroblert) [reply sample/full/cmd/flags.go]: done

File: flags.go (hunk 2 above)
Comment (chroblert) [reply flags.go]: done

File: flags.go (hunk 3 above)
Comment (chroblert) [reply flags.go]: done

File: flagmap.go (hunk 5 above)
Comment (chroblert) [reply flagmap.go]: done

**Comment by skaldesh** [comment]
Two small things:   
- Please rename `StringSlice` it flags.go to `StringList`. Then it conforms with arguments.  
- And we need a `StringListL()` method without a shorthand

**Comment by skaldesh** [comment]
handled in #43 now

**Pull request closed without merging**